### PR TITLE
Impl Dedicated Nasha

### PR DESCRIPTION
**Pull request by FlorianPerrot**
# Description

Partial implementation NASHA resources / data with test and docs.

Thanks @goblain. Author of first implementation. see : https://github.com/ovh/terraform-provider-ovh/pull/44

## Type of change

Add Resources :
- ovh_dedicated_nasha_partition
- ovh_dedicated_nasha_partition_access
- ovh_dedicated_nasha_partition_snapshot

Add Data:
- ovh_dedicated_nasha

### CODE REVIEW

**Comment by scraly** [comment]
Thanks :)
In every page of OVHcloud this service is called "HA-NAS" in english (or NAS-HA in french ;-) ).
So I added a suggestion of modification in all the PR (code and documentation) with this correct product name :).

File: ovh/resource_dedicated_nasha_partition.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	return nil
+}
+
+func resourceDedicatedNASHAPartitionRead(c context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+
+	if strings.Contains(d.Id(), "/") {
+		tab := strings.Split(d.Id(), "/")
+		if len(tab) != 2 {
+			return diag.Errorf("cant parse partition id: %s", d.Id())
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition.go:101]:
```suggestion
			return diag.Errorf("can't parse partition ID: %s", d.Id())
```

Comment (scraly) [inline ovh/resource_dedicated_nasha_partition.go:101]:
```suggestion
			return diag.Errorf("can't parse partition ID: %s", d.Id())
```

Comment (scraly) [inline ovh/resource_dedicated_nasha_partition.go:101]:
```suggestion
			return diag.Errorf("can't parse partition ID: %s", d.Id())
```

File: ovh/resource_dedicated_nasha_partition.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			return d, fmt.Sprint(resp.Size), nil
+		},
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(c)
+	if err != nil {
+		return diag.Errorf("waiting for NASHA partition update: %s", err)
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition.go:169]:
```suggestion
		return diag.Errorf("waiting for HA-NAS partition update: %s", err)
```

File: ovh/resource_dedicated_nasha_partition_access.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		return diag.Errorf("calling %s with params %v:\n\t %q", endpoint, access, err)
+	}
+
+	stateConf := resp.StateChangeConf(d, meta)
+
+	_, err = stateConf.WaitForStateContext(c)
+	if err != nil {
+		return diag.Errorf("waiting for NASHA partition access (%v): %s", access, err)
+	}
+	log.Printf("[DEBUG] Created NASHA partition access")
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_access.go:75]:
```suggestion
	log.Printf("[DEBUG] Created HA-NAS partition access")
```

File: ovh/resource_dedicated_nasha_partition_access.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	return nil
+}
+
+func resourceDedicatedNASHAPartitionAccessRead(c context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+
+	if strings.Contains(d.Id(), "/") {
+		tab := strings.Split(d.Id(), "/")
+		if len(tab) != 3 {
+			return diag.Errorf("cant parse access partition id: %s", d.Id())
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_access.go:88]:
```suggestion
			return diag.Errorf("can't parse access partition ID: %s", d.Id())
```

File: ovh/resource_dedicated_nasha_partition_snapshot.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+		return diag.Errorf("calling %s with params %q:\n\t %s", endpoint, snapshot, err.Error())
+	}
+
+	stateConf := resp.StateChangeConf(d, meta)
+
+	_, err = stateConf.WaitForStateContext(c)
+	if err != nil {
+		return diag.Errorf("waiting for NASHA snapshot (%q): %s", snapshot, err.Error())
+	}
+	log.Printf("[DEBUG] Created NASHA snapshot")
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_snapshot.go:66]:
```suggestion
	log.Printf("[DEBUG] Created HA-NAS snapshot")
```

File: ovh/resource_dedicated_nasha_test.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+		config := testAccProvider.Meta().(*Config)
+		endpoint := fmt.Sprintf("/dedicated/nasha/%s/partition/%s", serviceName, partitionName)
+		partitionResponse := &DedicatedNASHAPartition{}
+		err := config.OVHClient.Get(endpoint, partitionResponse)
+		if err != nil {
+			return fmt.Errorf("calling GET %s :\n\t %s", endpoint, err.Error())
+		}
+
+		fmt.Printf("NASHA partition : %+v\n", partitionResponse)
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_test.go:88]:
```suggestion
		fmt.Printf("HA-NAS partition: %+v\n", partitionResponse)
```

**Comment by FlorianPerrot** [comment]
Thanks @Scraly for review my PR.
I can replace on documentation and log message. But on "api.ovh.com" real entrypoint is called "/dedicated/nasha" to have a link between terraform resources and OVH resources i kept `dedicated_nasha`.

**Comment by scraly** [comment]
Yes the name of the resource and datasource are good, because it matches with the API endpoint but I think we need to be homogeneous with the product name: https://www.ovhcloud.com/en-gb/storage-solutions/nas-ha/ :)

File: ovh/resource_dedicated_nasha_test.go [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+func testAccNashaPartitionDestroy(n string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		nashaPartitionAccessResource, ok := state.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", n)
+		}
+
+		serviceName := nashaPartitionAccessResource.Primary.Attributes["service_name"]
+		parititionName := nashaPartitionAccessResource.Primary.Attributes["name"]
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_test.go:152]:
```suggestion
		partitionName := nashaPartitionAccessResource.Primary.Attributes["name"]
```

File: ovh/resource_dedicated_nasha_test.go [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+		config := testAccProvider.Meta().(*Config)
+		endpoint := fmt.Sprintf("/dedicated/nasha/%s/partition/%s", serviceName, parititionName)
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_test.go:155]:
```suggestion
		endpoint := fmt.Sprintf("/dedicated/nasha/%s/partition/%s", serviceName, partitionName)
```

Comment (scraly) [inline ovh/resource_dedicated_nasha_test.go:155]:
```suggestion
		endpoint := fmt.Sprintf("/dedicated/nasha/%s/partition/%s", serviceName, partitionName)
```

Comment (scraly) [inline ovh/resource_dedicated_nasha_test.go:155]:
```suggestion
		endpoint := fmt.Sprintf("/dedicated/nasha/%s/partition/%s", serviceName, partitionName)
```

**Review by scraly — CHANGES_REQUESTED** [review]
fix `parititionName` to `partitionName`

**Comment by scraly** [comment]
Thank you,
but some suggestions are not fixed for the moment :) 
image

**Comment by FlorianPerrot** [comment]
Fixed!
Sorry for this.

File: website/docs/r/dedicated_nasha_partition.markdown [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+# ovh_dedicated_nasha_partition
+
+Provides a resource for managing partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition" "foo" {
+  service_name = "zpool-12345"
+  name = "foo"
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:18]:
```suggestion
  name = "my-partition"
```

File: website/docs/r/dedicated_nasha_partition.markdown [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+* `name` - See Argument Reference above.
+* `size` - See Argument Reference above.
+* `protocol` - See Argument Reference above.
+* `capacity` - Reports stroage space useage in %.
+* `used_by_snapshots` - Reports storage space used by snapshots in %.
+
+## Import
+
+HA-NAS can be imported using the `{service_name}/{name}`, e.g.  
+`$ terraform import ovh_dedicated_nasha_partition.foo zpool-12345/foo`
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:47]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition.foo zpool-12345/my-partition`
```

File: website/docs/r/dedicated_nasha_partition_access.markdown [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+Provides a resource for managing access rights to partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition_access" "foo" {
+  service_name = "zpool-12345"
+  partition_name = "foo"
+  ip = "123.123.123.123/32"
+  protocol = "NFS"
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_access.markdown:20]:
```suggestion
  type = "readwrite"
```

File: website/docs/r/dedicated_nasha_partition_access.markdown [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+# ovh_dedicated_nasha_partition_access
+
+Provides a resource for managing access rights to partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition_access" "foo" {
+  service_name = "zpool-12345"
+  partition_name = "foo"
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_access.markdown:18]:
```suggestion
  partition_name = "my-partition"
```

File: website/docs/r/dedicated_nasha_partition_access.markdown [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+# ovh_dedicated_nasha_partition_access
+
+Provides a resource for managing access rights to partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition_access" "foo" {
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_access.markdown:16]:
```suggestion
resource "ovh_dedicated_nasha_partition_access" "my-partition" {
```

File: website/docs/r/dedicated_nasha_partition_access.markdown (hunk 12 above)
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_access.markdown:18]:
```suggestion
  partition_name = "my-partition"
```

File: website/docs/r/dedicated_nasha_partition.markdown [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+# ovh_dedicated_nasha_partition
+
+Provides a resource for managing partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition" "foo" {
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:16]:
```suggestion
resource "ovh_dedicated_nasha_partition" "my-partition" {
```

File: website/docs/r/dedicated_nasha_partition.markdown [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `service_name` - See Argument Reference above.
+* `name` - See Argument Reference above.
+* `size` - See Argument Reference above.
+* `protocol` - See Argument Reference above.
+* `capacity` - Reports stroage space useage in %.
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:41]:
```suggestion
* `capacity` - Reports storage space usage in %.
```

File: website/docs/r/dedicated_nasha_partition.markdown (hunk 14 above)
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:16]:
```suggestion
resource "ovh_dedicated_nasha_partition" "my-partition" {
```

File: website/docs/r/dedicated_nasha_partition_snapshot.markdown [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+# ovh_dedicated_nasha_partition_snapshot
+
+Provides a resource for managing **snapshot** to partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition_snapshot" "foo" {
+  service_name = "zpool-12345"
+  partition_name = "foo"
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:18]:
```suggestion
  partition_name = "my-partition"
```

File: website/docs/r/dedicated_nasha_partition_snapshot.markdown [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+# ovh_dedicated_nasha_partition_snapshot
+
+Provides a resource for managing **snapshot** to partitions on HA-NAS services
+
+## Example Usage
+
+```
+resource "ovh_dedicated_nasha_partition_snapshot" "foo" {
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:16]:
```suggestion
resource "ovh_dedicated_nasha_partition_snapshot" "my-partition" {
```

File: website/docs/r/dedicated_nasha_partition_snapshot.markdown [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+  partition_name = "foo"
+  type = "day-3"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `service_name` - (Required) The internal name of your HA-NAS (it has to be ordered via OVH interface)
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:27]:
```suggestion
* `service_name` - (Required) The internal name of your HA-NAS (it has to be ordered via OVHcloud interface)
```

Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:27]:
```suggestion
* `service_name` - (Required) The internal name of your HA-NAS (it has to be ordered via OVHcloud interface)
```

File: website/docs/r/dedicated_nasha_partition_snapshot.markdown [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+The following attributes are exported:
+
+* `service_name` - See Argument Reference above.
+* `partition_name` - See Argument Reference above.
+* `type` - See Argument Reference above.
+
+## Import
+
+HA-NAS partition snapshot can be imported using the `{service_name}/{partition_name}/{type}`, e.g.  
+`$ terraform import ovh_dedicated_nasha_partition_snapshot.foo zpool-12345/foo/day-3`
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:42]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition_snapshot.foo zpool-12345/my-partition/day-3`
```

**Review by scraly — CHANGES_REQUESTED** [review]
I added some suggestions again :-) 

Moreover, the creation of resources are working :

```
$ terraform apply
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxxxx/my-partition]
data.ovh_dedicated_nasha.foo: Read complete after 1s [id=zpool-xxxxxx]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # ovh_dedicated_nasha_partition_access.my-partition will be created
  + resource "ovh_dedicated_nasha_partition_access" "my-partition" {
      + id             = (known after apply)
      + ip             = "123.123.123.123/32"
      + partition_name = "my-partition"
      + service_name   = "zpool-xxxxxx"
      + type           = "readwrite"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

ovh_dedicated_nasha_partition_access.my-partition: Creating...
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [10s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [20s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [30s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [40s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [50s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Creation complete after 51s [id=zpool-xxxxxx/my-partition/123.123.123.123/32]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

But I have an issue when I want to "read" the partition_access :

```
$ terraform apply
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition_access.my-partition: Refreshing state... [id=zpool-xxxx/my-partition/123.123.123.123/32]
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxx/my-partition]
data.ovh_dedicated_nasha.foo: Read complete after 1s [id=zpool-xxxx]
╷
│ Error: cant parse access partition id: zpool-xxxx/my-partition/123.123.123.123/32
│
│   with ovh_dedicated_nasha_partition_access.my-partition,
│   on pr349.tf line 12, in resource "ovh_dedicated_nasha_partition_access" "my-partition":
│   12: resource "ovh_dedicated_nasha_partition_access" "my-partition" {
│
```

**Review by scraly — CHANGES_REQUESTED** [review]
I added some suggestions again :-) 

Moreover, the creation of resources are working :

```
$ terraform apply
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxxxx/my-partition]
data.ovh_dedicated_nasha.foo: Read complete after 1s [id=zpool-xxxxxx]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # ovh_dedicated_nasha_partition_access.my-partition will be created
  + resource "ovh_dedicated_nasha_partition_access" "my-partition" {
      + id             = (known after apply)
      + ip             = "123.123.123.123/32"
      + partition_name = "my-partition"
      + service_name   = "zpool-xxxxxx"
      + type           = "readwrite"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

ovh_dedicated_nasha_partition_access.my-partition: Creating...
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [10s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [20s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [30s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [40s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [50s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Creation complete after 51s [id=zpool-xxxxxx/my-partition/123.123.123.123/32]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

But I have an issue when I want to "read" the partition_access :

```
$ terraform apply
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition_access.my-partition: Refreshing state... [id=zpool-xxxx/my-partition/123.123.123.123/32]
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxx/my-partition]
data.ovh_dedicated_nasha.foo: Read complete after 1s [id=zpool-xxxx]
╷
│ Error: cant parse access partition id: zpool-xxxx/my-partition/123.123.123.123/32
│
│   with ovh_dedicated_nasha_partition_access.my-partition,
│   on pr349.tf line 12, in resource "ovh_dedicated_nasha_partition_access" "my-partition":
│   12: resource "ovh_dedicated_nasha_partition_access" "my-partition" {
│
```

File: ovh/resource_dedicated_nasha_partition_access.go [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+
+	return nil
+}
+
+func resourceDedicatedNASHAPartitionAccessRead(c context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+
+	if strings.Contains(d.Id(), "/") {
+		tab := strings.Split(d.Id(), "/")
+		if len(tab) != 3 {
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_access.go:87]: a little fix? :-)

```suggestion
		if len(tab) != 4 {
```

Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_access.go:87]: a little fix? :-)

```suggestion
		if len(tab) != 4 {
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by FlorianPerrot** [comment]
I fixed this issue yesterday. here: https://github.com/ovh/terraform-provider-ovh/pull/349/commits/9940526a4697f21818c972381a80364e566abd66
Can you retry your test with this fix?

Acceptance test case works for me. If I'm not mistaken, he should also work for your test.

File: ovh/resource_dedicated_nasha_partition_access.go [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+
+		d.Set("service_name", tab[0])
+		d.Set("partition_name", tab[1])
+		ip, _ := url.PathUnescape(tab[2])
+		d.Set("ip", ip)
+	}
+
+	serviceName := d.Get("service_name").(string)
+	partitionName := d.Get("partition_name").(string)
+	ipsubnet, _ := d.Get("ip").(string)
```
Comment (scraly) [inline ovh/resource_dedicated_nasha_partition_access.go:99]: The IP stored/read is not correct.

Indeed, when I apply an access resource:

```
ovh_dedicated_nasha_partition_access.my-partition: Creating...
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [10s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [20s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [30s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [40s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [50s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Still creating... [1m0s elapsed]
ovh_dedicated_nasha_partition_access.my-partition: Creation complete after 1m1s [id=zpool-xxxx/my-partition/123.123.123.123/32]

Apply complete! Resources: 2 added, 0 changed, 1 destroyed.
```

And when I do an apply again (after solved a little the read function):

```
 $ terraform apply
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxxx/my-partition]
ovh_dedicated_nasha_partition_access.my-partition: Refreshing state... [id=zpool-xxxxxxx/my-partition/123.123.123.123/32]
ovh_dedicated_nasha_partition_snapshot.my-partition: Refreshing state... [id=zpool-xxxxxx/my-partition/day-3]
data.ovh_dedicated_nasha.foo: Read complete after 1s [id=zpool-xxxxx]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # ovh_dedicated_nasha_partition_access.my-partition must be replaced
-/+ resource "ovh_dedicated_nasha_partition_access" "my-partition" {
      ~ id             = "zpool-xxxx/my-partition/123.123.123.123/32" -> (known after apply)
      ~ ip             = "123.123.123.123" -> "123.123.123.123/32" # forces replacement
        # (3 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: ^C

Interrupt received.
Please wait for Terraform to exit or data loss may occur.
Gracefully shutting down...

╷
│ Error: error asking for approval: interrupted
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by scraly** [comment]
> I fixed this issue yesterday. here: [9940526](https://github.com/ovh/terraform-provider-ovh/commit/9940526a4697f21818c972381a80364e566abd66) Can you retry your test with this fix?
> 
> Acceptance test case works for me. If I'm not mistaken, he should also work for your test.

After your fix, I have still this issue in the read access:

```
$ terraform destroy
data.ovh_dedicated_nasha.foo: Reading...
ovh_dedicated_nasha_partition_snapshot.my-partition: Refreshing state... [id=zpool-xxxxx/my-partition/day-3]
ovh_dedicated_nasha_partition_access.my-partition: Refreshing state... [id=zpool-xxxxx/my-partition/123.123.123.123/32]
ovh_dedicated_nasha_partition.my-partition: Refreshing state... [id=zpool-xxxxx/my-partition]
data.ovh_dedicated_nasha.foo: Read complete after 0s [id=zpool-xxxxx]
╷
│ Error: can't parse access partition ID: zpool-xxxxx/my-partition/123.123.123.123/32
│
│   with ovh_dedicated_nasha_partition_access.my-partition,
│   on pr349.tf line 12, in resource "ovh_dedicated_nasha_partition_access" "my-partition":
│   12: resource "ovh_dedicated_nasha_partition_access" "my-partition" {
│
```

**Comment by FlorianPerrot** [comment]
You should retry without context. In your case the access resource have a bad ID, generated before my fix.

File: website/docs/d/dedicated_nasha.html.markdown [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+# ovh_dedicated_nasha (Data Source)
+
+Use this data source to retrieve information about a dedicated HA-NAS.
+
+## Example Usage
+
+```hcl
+data "ovh_dedicated_nasha" "foo" {
```
Comment (scraly) [inline website/docs/d/dedicated_nasha.html.markdown]:
```suggestion
data "ovh_dedicated_nasha" "my-nas-ha" {
```

File: website/docs/r/dedicated_nasha_partition.markdown [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+* `used_by_snapshots` - Reports storage space used by snapshots in %.
+
+## Import
+
+HA-NAS can be imported using the `{service_name}/{name}`, e.g.  
+`$ terraform import ovh_dedicated_nasha_partition.foo zpool-12345/my-partition`
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:49]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition.my-partition zpool-12345/my-partition`
```

Comment (scraly) [inline website/docs/r/dedicated_nasha_partition.markdown:49]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition.my-partition zpool-12345/my-partition`
```

File: website/docs/r/dedicated_nasha_partition_access.markdown [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+
+* `service_name` - See Argument Reference above.
+* `partition_name` - See Argument Reference above.
+* `ip` - See Argument Reference above.
+* `type` - See Argument Reference above.
+
+## Import
+
+HA-NAS partition access can be imported using the `{service_name}/{partition_name}/{ip}`, e.g.  
+`$ terraform import ovh_dedicated_nasha_partition_access.foo zpool-12345/my-partition/123.123.123.123%2F32`
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_access.markdown:45]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition_access.my-partition zpool-12345/my-partition/123.123.123.123%2F32`
```

File: website/docs/r/dedicated_nasha_partition_snapshot.markdown [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+The following attributes are exported:
+
+* `service_name` - See Argument Reference above.
+* `partition_name` - See Argument Reference above.
+* `type` - See Argument Reference above.
+
+## Import
+
+HA-NAS partition snapshot can be imported using the `{service_name}/{partition_name}/{type}`, e.g.  
+`$ terraform import ovh_dedicated_nasha_partition_snapshot.foo zpool-12345/my-partition/day-3`
```
Comment (scraly) [inline website/docs/r/dedicated_nasha_partition_snapshot.markdown:42]:
```suggestion
`$ terraform import ovh_dedicated_nasha_partition_snapshot.my-partition zpool-12345/my-partition/day-3`
```

**Review by scraly — CHANGES_REQUESTED** [review]
Tests are now Ok for me :-)

So please look at every suggestions have done and commit them.
Thank you

**Review by scraly — APPROVED** [review]

**Comment by scraly** [comment]
Good new feature, it's OK for me :)

**Pull request merged**